CMakeLists.txt

```diff
@@ -12,8 +24,25 @@ option(BUILD_TESTS "${PROJECT_NAME} - Build tests" ON)
 option(WITH_PROFILING "${PROJECT_NAME} - Build with profiling flags" OFF)
 option(WITH_C_BINDINGS "Build iron C bindings" ON)
 option(WITH_Python_BINDINGS "Build iron Python bindings" ON)
+CMAKE_DEPENDENT_OPTION(WITH_DIAGNOSTICS "Build iron with diagnostics support" ON
+                       "IS_DEBUG_BUILD" OFF)
+option(PRINT_CONFIG_SUMMARY "Show a summary of the configuration." TRUE)
 set(PACKAGE_CONFIG_DIR "cmake" CACHE STRING "Directory for package config files (relative to CMAKE_INSTALL_PREFIX)")
 
+string(TOUPPER "${REAL_PRECISION}" _uppercase_REAL_PRECISION)
+if(_uppercase_REAL_PRECISION STREQUAL "SINGLE")
+  set(WORKING_REAL_PRECISION_SINGLE)
+elseif(_uppercase_REAL_PRECISION STREQUAL "DOUBLE")
+  set(WORKING_REAL_PRECISION_DOUBLE)
```

Comment:
Yes, although there may be quad later and so I was trying to avoid just having a binary option.